Source/CBL_Replicator.m

```diff
@@ -341,13 +341,6 @@ - (void) start {
                  }
                 ];
 
-    // If client didn't set an authorizer, use basic auth if credential is available:
-    if (!_authorizer) {
-        _authorizer = [[CBLBasicAuthorizer alloc] initWithURL: _remote];
```

Comment:
I think we need to, because we need an authorizer to handle digest auth, and because the two are pretty interchangeable except for the details of what goes in the Authorization header. Probably should be renamed CBLPasswordAuthorizer.

You're right that we shouldn't send the username/password before receiving the 401 authentication challenge from the server. We'll also need to improve the CBLAuthorizer API so the implementation can see the challenge info when responding — at a minimum it should be given the value of the response's WWW-Authenticate header.